code/modules/atmospherics/gasmixtures/reactions.dm

```diff
@@ -193,44 +193,43 @@
 	min_requirements = list(
 		"ENER" = PLASMA_BINDING_ENERGY * 10,
 		/datum/gas/plasma = MINIMUM_HEAT_CAPACITY,
-		/datum/gas/tritium = MINIMUM_HEAT_CAPACITY
+		/datum/gas/carbon_dioxide = 1
 	)
 
 /datum/gas_reaction/fusion/react(datum/gas_mixture/air, turf/open/location)
 	var/list/cached_gases = air.gases
 	var/temperature = air.temperature
-
-	if(((cached_gases[/datum/gas/plasma][MOLES]+cached_gases[/datum/gas/tritium][MOLES])/air.total_moles() < FUSION_PURITY_THRESHOLD) || air.return_pressure() < 10*ONE_ATMOSPHERE)
-		//Fusion wont occur if the level of impurities is too high or if there is too little pressure.
+	if(air.return_pressure() < 10*ONE_ATMOSPHERE)
+		//Fusion wont occur if there is too little pressure.
 		return NO_REACTION
 
 	var/old_heat_capacity = air.heat_capacity()
-	var/catalyst_efficency = max(min(cached_gases[/datum/gas/plasma][MOLES]/cached_gases[/datum/gas/tritium][MOLES],MAX_CATALYST_EFFICENCY)-(temperature/FUSION_HEAT_DROPOFF),1)
 	var/reaction_energy = THERMAL_ENERGY(air)
-	var/moles_impurities = air.total_moles()-(cached_gases[/datum/gas/plasma][MOLES]+cached_gases[/datum/gas/tritium][MOLES])
-
-	var/plasma_fused = (PLASMA_FUSED_COEFFICENT*catalyst_efficency)*(temperature/PLASMA_BINDING_ENERGY)/10
-	var/tritium_catalyzed = (CATALYST_COEFFICENT*catalyst_efficency)*(temperature/PLASMA_BINDING_ENERGY)/40
-	var/oxygen_added = tritium_catalyzed
-	var/waste_added = max((plasma_fused-oxygen_added)-((air.total_moles()*air.heat_capacity())/PLASMA_BINDING_ENERGY),0)
-	reaction_energy = max(reaction_energy+((catalyst_efficency*cached_gases[/datum/gas/plasma][MOLES])/((moles_impurities/catalyst_efficency)+2)*10)+((plasma_fused/((moles_impurities/catalyst_efficency)))*PLASMA_BINDING_ENERGY),0)
-
-	air.assert_gases(/datum/gas/oxygen, /datum/gas/carbon_dioxide, /datum/gas/water_vapor, /datum/gas/nitrous_oxide, /datum/gas/nitryl)
-	//Fusion produces an absurd amount of waste products now, requiring active filtration.
-	cached_gases[/datum/gas/plasma][MOLES] = max(cached_gases[/datum/gas/plasma][MOLES] - plasma_fused,0)
-	cached_gases[/datum/gas/tritium][MOLES] = max(cached_gases[/datum/gas/tritium][MOLES] - tritium_catalyzed,0)
-	cached_gases[/datum/gas/oxygen][MOLES] += oxygen_added
-	cached_gases[/datum/gas/water_vapor][MOLES] += waste_added
-	cached_gases[/datum/gas/nitrous_oxide][MOLES] += waste_added
-	cached_gases[/datum/gas/nitryl][MOLES] += waste_added
-	cached_gases[/datum/gas/carbon_dioxide][MOLES] += waste_added
-	if (location)
-		radiation_pulse(location, reaction_energy/(PLASMA_BINDING_ENERGY*MAX_CATALYST_EFFICENCY))
+	var/mediation = (air.heat_capacity()-(cached_gases[/datum/gas/plasma][MOLES]*cached_gases[/datum/gas/plasma][GAS_META][META_GAS_SPECIFIC_HEAT]))/(air.total_moles()-cached_gases[/datum/gas/plasma][MOLES])
```

Comment:
this is temperate in the gases other than plasma, correct? could you indicate that with a comment?

Reply:
Average heat capacity not including plasma, but yeah I will.

Reply:
Clamp this to prevent negative numbers and zeros, you don't want division to fuck up

Reply:
Those shouldn't be possible, since that would either requiring total_moles to be less than the amount of plasma moles(impossible, equal at most), or require them to be the same, which would cause the reaction to not meet its minimum requirement for carbon dioxide for the reaction to occur.

Reply:
Its just that this has been comically broken before, the math on this one is hard to follow across the 4 pages

Reply:
While fusion does have an impressive tendency to break in crazy ways, I think that even if it's possible for a bug like that to occur here, we're better off seeing it break then hiding it with a clamp or a max. Since that would imply either reaction checking is broken(pretty bad) or total_moles() is broken(REALLY bad). As it stands though I think it's fine as it is

Reply:
this formula runtiming would be my worst nightmare lol, this is quite safe